### PR TITLE
Implement `koji_build` trigger for `koji_build` job

### PR DESCRIPTION
**Pull request by nforro**
Fixes https://github.com/packit/packit-service/issues/2378.

Merge after https://github.com/packit/packit/pull/2349.
Merge after https://github.com/packit/packit/pull/2350.

### CODE REVIEW

File: alembic/versions/e05e1b04de87_add_koji_build_tags.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        ["koji_tag_name"],
+        unique=False,
+    )
+    op.create_index(
+        op.f("ix_koji_build_tags_project_id"),
+        "koji_build_tags",
+        ["project_id"],
+        unique=False,
+    )
+    op.execute("ALTER TYPE projecteventtype ADD VALUE 'koji_build_tag'")
```
Comment (nforro) [inline alembic/versions/e05e1b04de87_add_koji_build_tags.py:49]: There is no easy way to remove a value from an enum type, so there is no corresponding reverse operation in `downgrade()`. Could that be a problem?

**Comment by nforro** [comment]
recheck

**Comment by nforro** [comment]
recheck

**Comment by nforro** [comment]
recheck

**Comment by nforro** [comment]
recheck

File: alembic/versions/e05e1b04de87_add_koji_build_tags.py (hunk 1 above)
Comment (lbarcziova) [reply alembic/versions/e05e1b04de87_add_koji_build_tags.py:49]: I don't think so, I think in the past we did the same for other migrations

**Comment by nforro** [comment]
recheck

File: packit_service/worker/handlers/koji.py [hunk 2]
```diff
@@ -354,3 +366,79 @@ def run(self):
...
+        packages_to_trigger = set()
+        for package_name in tagged_packages:
+            distgit_project_url = f"{dg_base_url}rpms/{package_name}"
+            project = self.service_config.get_project(url=distgit_project_url)
+            packages_config = PackageConfigGetter.get_package_config_from_repo(
+                base_project=None,
+                project=project,
+                pr_id=None,
+                reference=None,
+                fail_when_missing=True,
```
Comment (lbarcziova) [inline packit_service/worker/handlers/koji.py:407]: do we want to have this `True`? maybe we should handle this case in a different way?

**Review by lbarcziova — COMMENTED** [review]
very nice work!

File: packit_service/worker/handlers/koji.py (hunk 2 above)
Comment (nforro) [reply packit_service/worker/handlers/koji.py:407]: Right, I didn't realize it's possible for builds of packages without Packit config to be tagged into the sidetag, so in this case skipping the package would be certainly better, same below. Let me change that.

**Review by lbarcziova — APPROVED** [review]
LGTM!

**Pull request merged**